es-app/src/SystemScreenSaver.cpp

```diff
@@ -301,8 +301,8 @@ void SystemScreenSaver::pickGameListNode(unsigned long index, const char *nodeNa
 		pugi::xml_document doc;
 		pugi::xml_node root;
 
-		// We only want images from game systems
-		if (!(*it)->isGameSystem())
+		// We only want nodes from game systems that are not collections
+		if (!(*it)->isGameSystem() || (*it)->isCollection())
```

Comment:
In the countGameListNodes() function (used in countVideos() and countImages()), we use:

  if (!(*it)->isCollection() && (*it)->isGameSystem())

to make sure that the node is NOT a collection and IS a game system.

In pickGameListNode(), we use:

  if (!(*it)->isGameSystem() || (*it)->isCollection())

BUT we "continue" (skip over the node) if that matches.  So if the node is NOT a game system, or it IS a collection, then skip the node.

I was having a problem where the Slideshow Screensaver was showing the icons from the "retropie" system, so I decided to add the "isGameSystem()" checks.

I also want to make sure that the checks are consistent when counting the media files, and when deciding if the node is the "random" one.  I don't want to ignore collections when counting media, but use them when actually picking the node.  Honestly, I haven't played around with collections at all, so I don't really understand how they work.  I assume that they are a "virtual game set"- a subset of a real system's games...?